### PR TITLE
KAFKA-5931: deprecate KTable#through and KTable#to

### PR DESCRIPTION
**Pull request by dguy**

### CODE REVIEW

**Comment by dguy** [comment]
@guozhangwang, @bbejeck, @mjsax

**Comment by bbejeck** [comment]
Thanks, @dguy just one meta comment. Do want to add `toStream()#through` or `toStream()#to` in the deprecated description?  Otherwise LGTM.

**Comment by dguy** [comment]
@bbejeck problem is the link tags don't work like that, so it would have to be `{@code toStream().through()}` etc

**Comment by guozhangwang** [comment]
retest this please

**Comment by guozhangwang** [comment]
LGTM. Feel free to merge after Jenkins passed.

File: streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java [hunk 1]
```diff
@@ -763,17 +763,19 @@ void writeAsText(final String filePath,
...
      * The resulting {@code KTable} will be materialized in a local state store with the given store name (cf.
-     * {@link StreamsBuilder#table(String, String)})
+     * {@link StreamsBuilder#table(String, Materialized)})
      * The store name must be a valid Kafka topic name and cannot contain characters other than ASCII alphanumerics, '.', '_' and '-'.
      *
      * @param topic     the topic name
      * @param queryableStoreName the state store name used for the result {@code KTable}; valid characters are ASCII
      *                  alphanumerics, '.', '_' and '-'. If {@code null} this is the equivalent of {@link KTable#through(String)()}
      * @return a {@code KTable} that contains the exact same (and potentially repartitioned) records as this {@code KTable}
+     * @deprecated use {@link #toStream()}
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java]: `use {@link #toStream()} followed by {@link KStream#to(String)} and {@link StreamsBuilder#table(String)} to read back as a {@code KTable}` ??

same below

**Review by mjsax — COMMENTED** [review]
For `to()` JavaDoc should be different to my suggestion of course.

**Comment by dguy** [comment]
merged to trunk

**Pull request closed without merging**